Sources/Fuzzilli/FuzzIL/TypeSystem.swift

```diff
@@ -850,53 +789,168 @@ class TypeExtension: Hashable {
     }
 }
 
+public enum Parameter: Equatable, Hashable {
+    case plain(Type)
+    case opt(Type)
+    case rest(Type)
+
+    fileprivate func format(abbreviate: Bool) -> String {
+        switch self {
+            case .plain(let t):
+                return ".plain(\(t.format(abbreviate: abbreviate)))"
+            case .opt(let t):
+                return ".opt(\(t.format(abbreviate: abbreviate)))"
+            case .rest(let t):
+               return ".rest(\(t.format(abbreviate: abbreviate)))"
+        }
+    }
+
+    fileprivate func callerType() -> Type {
+        switch self {
+            case .plain(let t):
+                return t
+            case .opt(let t):
+                return t
+            case .rest(_):
+               return .iterable
```

Comment:
The caller doesn't pass an iterable, it just passes potentially multiple arguments of type `t`. So let's just return `t` here as well